service/src/java/org/apache/hive/service/cli/operation/HiveCommandOperation.java

```diff
@@ -124,14 +139,45 @@ public void runInternal() throws HiveSQLException {
       }
     } catch (CommandProcessorException e) {
       setState(OperationState.ERROR);
-      throw toSQLException("Error while processing statement", e);
+      throw toSQLException("Error while processing statement: " + e.toString(), e);
```

Comment:
Why do we need to type cast the exception to string? We are anyway extracting message, Sqlstate, and response code from the exception.

Reply:
Thanks!  Yes, it is not needed.